foxglove_data_platform/client.py

```diff
@@ -281,12 +292,20 @@ def get_messages(
         topics: An optional list of topics to retrieve.
             All topics will be retrieved if this is omitted.
         """
+        if not device_id or device_name:
```

Comment:
```suggestion
        if device_id is None and device_name is None:
```

Lots of different objects are falsy in python, I prefer to compare to None directly in case there's a real device named `""`.